packages/snowpack/src/build/build-pipeline.ts

```diff
@@ -63,12 +68,40 @@ async function runPipelineLoadStep(
       const mainOutputExt = step.resolve.output[0];
       return {[mainOutputExt]: result};
     } else if (result && typeof result === 'object') {
+      result;
+
+      // handle source maps
+      Object.keys(result).forEach((ext) => {
+        const output = result[ext];
+        if (typeof output !== 'object' || !output.code) return;
+
+        if (output.map) {
+          // some plugins return map as an object; others as a string. handle both cases
+          result[ext + '.map'] =
+            typeof output.map === 'object' ? JSON.stringify(output.map) : output.map;
+
+          const sourceMapFile = path
+            .basename(srcPath)
+            .replace(new RegExp(srcExt + '$', 'i'), ext + '.map');
+
+          // Source Map Spec v3: https://docs.google.com/document/d/1U1RGAehQwRypUTovF1KRlpiOFze0b-_2gc6fAH0KY0k/edit#heading=h.lmz475t4mvbx
+          if (ext === '.css') {
+            output.code += `/*# sourceMappingURL=${sourceMapFile} */`;
+          } else {
+            output.code += `\n//# sourceMappingURL=${sourceMapFile}\n`; // add newline at beginning & end
+            output.code = output.code.replace(/\n*$/, '\n'); // remove extra lines at EOF
+          }
+        }
+      });
```

Comment:
This is where we handle the new, optional `{ code: string, map: string | object }` syntax for plugins.